src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, healt) {
+    super(name, healt);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+}
+
+if (Herbivore.health <= 0) {
+  Animal.alive = Animal.alive.filter(animal => animal.health > 0);
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore.hidden === false) {
```

Comment:
`!herbivore.hidden` it's the same as `herbivore.hidden === false`
```suggestion
    if (!herbivore.hidden) {
```